kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/util/LinuxNetworkUtil.java

```diff
@@ -264,22 +269,45 @@ private boolean isWifiLinkUpInternal(String ifaceName) throws KuraException {
     }
 
     public static boolean toolExists(String tool) {
-        boolean ret = false;
-        final String[] searchFolders = new String[] { "/sbin/", "/usr/sbin/", "/bin/", "/usr/bin/" };
-
         if (TOOLS.contains(tool)) {
-            ret = true;
+            return true;
         } else {
-            for (String folder : searchFolders) {
-                File fTool = new File(folder + tool);
-                if (fTool.exists()) {
+            for (String folder : DEFAULT_PATH) {
+                File fUnit = new File(folder + tool);
+                if (fUnit.exists()) {
                     TOOLS.add(tool);
-                    ret = true;
-                    break;
+                    return true;
                 }
             }
         }
-        return ret;
+        return false;
+    }
+
+    /**
+     * Checks if the given Systemd system unit is installed.
+     * The result is based on the exit code of "systemctl status <unitName>"
+     * as presented in https://www.man7.org/linux/man-pages/man1/systemctl.1.html#EXIT_STATUS
+     *
+     * @param unitName
+     *            the name of the Systemd system unit
+     * @return true if the unit is installed
+     */
+    public static boolean systemdSystemUnitExists(String unitName) {
+        PROCESS_BUILDER.command("sh", "-c", "systemctl status " + unitName);
```

Comment:
I already know there's a reason for these but, just we are all on the same page and I can learn something new, can you explain why can't you use the instance's `CommandExecutorService`?

Reply:
The [CommandExecutorService](https://github.com/eclipse-kura/kura/blob/develop/kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/executor/CommandExecutorService.java) is an interface that is extended by the [PrivilegedExecutorService](https://github.com/eclipse-kura/kura/blob/develop/kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/executor/PrivilegedExecutorService.java) and the [UnprivilegedExecutorService](https://github.com/eclipse-kura/kura/blob/develop/kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/executor/UnprivilegedExecutorService.java) as marker interfaces. 

They are implemented by the [PrivilegedExecutorServiceImpl](https://github.com/eclipse-kura/kura/blob/develop/kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/linux/executor/privileged/PrivilegedExecutorServiceImpl.java) and [UnprivilegedExecutorServiceImpl](https://github.com/eclipse-kura/kura/blob/develop/kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/linux/executor/unprivileged/UnprivilegedExecutorServiceImpl.java).

The idea behind this classes is to provide a way to run system command using privileged (e.g. kurad) or unprivileged (e.g. kura) linux users. They are both OSGI services and are activated during startup.

The `LinuxNetworkUtil` class is a standard Java class and it is not registered in the OSGI framework. It cannot be bounded or use other services. Indeed, the `CommandExecutorService` is passed as parameter to the constructor, since some methods need it. The static methods cannot use it since the passed reference to the service is not static. Moreover, the service cannot be passed to the static methods of this class, since the caller (e.g. in the `DhcpServermanager`) is static too and it haven't access to the `CommandExecutorService` instantiated in that class.

One way to overcome this is to directly instantiate a `PrivilegedCommandExecutorServiceImpl`, but I refuse to do it.